Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs

```diff
@@ -18,16 +21,109 @@ public sealed partial class VendingMachineMenu : DefaultWindow
 
         public VendingMachineMenu()
         {
-            MinSize = SetSize = (250, 150);
             RobustXamlLoader.Load(this);
             IoCManager.InjectDependencies(this);
 
             VendingContents.OnItemSelected += args =>
             {
-                OnItemSelected?.Invoke(args);
+                // OnItemSelected?.Invoke(args);
+                onItemSelected(args);
             };
+
+            VendButton.OnPressed += _ =>
+            {
+                // Stupid solution... but it works.
+
+                int pos = 0;
+                int finalPos = 0;
+                foreach (var item in VendingContents)
+                {
+                    if (item.Selected)
+                    {
+                        finalPos = pos;
+                        pos++;
+                        break;
+                    }
+                    pos++;
+                }
+
+                if (finalPos != 0 && finalPos < pos) OnItemSelected?.Invoke(new ItemList.ItemListSelectedEventArgs(finalPos, VendingContents));
+            };
+
+
+            // Placeholder VendingInfo
+            var entityManager = IoCManager.Resolve<IEntityManager>();
+            var ent = entityManager.SpawnEntity("VendingMachineCola", MapCoordinates.Nullspace);
```

Comment:
Any reason you don't use a spriteview of the entity?

You also never cleaned this entity up.

Reply:
it is using spriteview?

Reply:
it is using spriteview